python/models/hierarchical_filtering_proxy_model.py

```diff
@@ -141,7 +142,7 @@ def minimize(self):
                 return
 
             self._cache = dict(
-                [(k, v) for k, v in self._cache.iteritems() if v[0].isValid()]
+                [(k, v) for k, v in six.iteritems(self._cache) if v[0].isValid()]
```

Comment:
Filtering model need to be super-fast, so I'm keeping the use of iterators here.